### PR TITLE
Add length validation to text input

### PR DESCRIPTION
**Pull request by bpedersen2**
This change allows to enforce minimum and maximum length
limit and/or pattern restrictions for regform text input fields.

### CODE REVIEW

File: indico/htdocs/js/indico/modules/registration/form/tpls/fields/dialogs/size.tpl.html [hunk 1]
```diff
@@ -9,3 +9,32 @@
...
+            >
+        <i class="info-helper" title="{{ 'Minimimum  allowed length' | i18n }}"></i>
+    </div>
+    <div class="regFormEditLine">
+        <label class="regFormDialogCaption">
+            {{ 'Validation pattern' | i18n }}
+        </label>
+        <input type="text" name="validpattern"
+            ng-model="formData.validPattern"
+            >
```
Comment (ThiefMaster) [inline indico/htdocs/js/indico/modules/registration/form/tpls/fields/dialogs/size.tpl.html]: should go in the previous line (also two more times above)

File: indico/htdocs/js/indico/modules/registration/form/tpls/fields/dialogs/size.tpl.html [hunk 2]
```diff
@@ -9,3 +9,32 @@
...
+        <i class="info-helper" title="{{ 'Validation pattern for input , see https://www.w3schools.com/jsref/jsref_obj_regexp.asp }}' | i18n }}"></i>
```
Comment (ThiefMaster) [inline indico/htdocs/js/indico/modules/registration/form/tpls/fields/dialogs/size.tpl.html]: Even though w3schools isn't as awful as they used to be (but the still sell those phony "html certificates"), I don't want a link to them in Indico ;) (also, there shouldn't be a space before the `,`).

Maybe letting users specify regex patterns here is a bit overkill anyway...

File: indico/htdocs/js/indico/modules/registration/form/tpls/fields/dialogs/size.tpl.html [hunk 3]
```diff
@@ -9,3 +9,32 @@
...
 </div>
+<ng-if="field.inputType == 'text'>
+    <div class="regFormEditLine">
+        <label class="regFormDialogCaption">
+            {{ 'Max length in char' | i18n }}
+        </label>
+        <input type="number" name="maxlength"
+            ng-model="formData.maxLength"
+            >
+        <i class="info-helper" title="{{ 'Maximimum  allowed length' | i18n }}"></i>
```
Comment (ThiefMaster) [inline indico/htdocs/js/indico/modules/registration/form/tpls/fields/dialogs/size.tpl.html]: to many spaces

File: indico/htdocs/js/indico/modules/registration/form/tpls/fields/dialogs/size.tpl.html (hunk 1 above)
Comment (bpedersen2) [reply indico/htdocs/js/indico/modules/registration/form/tpls/fields/dialogs/size.tpl.html]: done

File: indico/htdocs/js/indico/modules/registration/form/tpls/fields/dialogs/size.tpl.html (hunk 3 above)
Comment (bpedersen2) [reply indico/htdocs/js/indico/modules/registration/form/tpls/fields/dialogs/size.tpl.html]: done

File: indico/htdocs/js/indico/modules/registration/form/tpls/fields/dialogs/size.tpl.html (hunk 2 above)
Comment (bpedersen2) [reply indico/htdocs/js/indico/modules/registration/form/tpls/fields/dialogs/size.tpl.html]: removed the link. 

Using pattern validation can e.g. be used if you need special input from the registrants (e.g only uppercaser, no spaces, etc.)

File: indico/htdocs/js/indico/modules/registration/form/tpls/fields/dialogs/size.tpl.html [hunk 4]
```diff
@@ -9,3 +9,26 @@
...
+        </label>
+        <input type="number" name="minlength" ng-model="formData.minLength">
+        <i class="info-helper" title="{{ 'Minimimum allowed length' | i18n }}"></i>
+    </div>
+    <div class="regFormEditLine">
+        <label class="regFormDialogCaption">
+            {{ 'Validation pattern' | i18n }}
+        </label>
+        <input type="text" name="validpattern" ng-model="formData.validPattern">
+        <i class="info-helper" title="{{ 'Validation pattern(a javascript regualr expression) }}' | i18n }}"></i>
```
Comment (ThiefMaster) [inline indico/htdocs/js/indico/modules/registration/form/tpls/fields/dialogs/size.tpl.html]: missing space before `(` and a typo (regualr).

Anyway, I still don't like the idea of having the pattern option. Yes, it is powerful, but eventually we want server-side validation for this stuff and except when restricting yourself to a common subset of the regex features of HTML5/JS and Python, that won't be guaranteed to work on both sides...

Also, having user-provided regexps run on the server may be risky - usually there are some ways to write regular expressions that result in [horrible performance](http://www.regular-expressions.info/catastrophic.html).

File: indico/htdocs/js/indico/modules/registration/form/tpls/fields/text.tpl.html [hunk 5]
```diff
@@ -5,10 +5,25 @@
...
         ng-model="userdata[fieldName]"
         ng-required="field.isRequired"
+        ng-maxlength="{{ field.maxLength }}"
+        ng-minlength="{{ field.minLength }}"
+        ng-pattern="{{ field.validPattern }}"
         ng-class="{
             hasError: validationStarted && nestedForm.$invalid
         }"/>
 
+    <span ng-show="validationStarted && nestedForm.$invalid && nestedForm.$error.minlength"  class="error-message-tag">
```
Comment (ThiefMaster) [inline indico/htdocs/js/indico/modules/registration/form/tpls/fields/text.tpl.html]: double space before `class` (same below)

Comment (bpedersen2) [reply indico/htdocs/js/indico/modules/registration/form/tpls/fields/text.tpl.html]: fixed

File: indico/htdocs/js/indico/modules/registration/form/tpls/fields/dialogs/size.tpl.html (hunk 4 above)
Comment (bpedersen2) [reply indico/htdocs/js/indico/modules/registration/form/tpls/fields/dialogs/size.tpl.html]: I splitted this part out, as regexp validation is not  too important for now.

**Comment by bpedersen2** [comment]
Ping...

**Comment by ThiefMaster** [comment]
Don't worry, we didn't forget about it - just busy with other things ;)

It's cleaned up (typos etc.) and merged now.

**Pull request merged**